api-models/course.yaml

```
@@ -0,0 +1,154 @@
+openapi: 3.0.0
+info:
+  title: Courses Microservice API
+  description: This API manages different types of courses in SkillVerse
+  version: 1.0.0
+
+paths:
+  /courses:
+    get:
+      summary: List all courses
+      operationId: listCourses
+      parameters: 
+        - name: only
+          in: query
+          description: Array of course IDs to fetch a subset of courses
+          required: false
+          style: form
+          explode: true
+          schema:
+            type: array
+            items:
+              type: string
+        - name: filterByTag
+          in: query
+          description: Array of tag IDs to filter the courses by tags
+          required: false
+          explode: true
+          schema:
+            type: array
+            items:
+              type: string
+        - name: filterByTitle
+          in: query
+          description: String which should be part of the title
+          required: false
+          explode: true
+          schema:
+            type: string
+        - name: nested
+          in: query
+          description: Determines if the nested data is fetched
```

```suggestion
          description: Determines if the nested data will be returned
```
Is this a more fitting description? I'm not sure